serveradmin/serverdb/models.py

```diff
@@ -573,6 +668,28 @@ class Meta:
         unique_together = [['server', 'attribute', 'value']]
         index_together = [['attribute', 'value']]
 
+    def clean(self):
+        super(ServerAttribute, self).clean()
+
+        # Get the ip_addr_type of the servertype
+        ip_addr_type = self.server.servertype.ip_addr_type
+
+        if ip_addr_type == 'null':
+            # A Servertype with ip_addr_type "null" and attributes of type
+            # inet must be denied per configuration. This is just a safety net
+            # in case e.g. somebody creates them programmatically.
+            raise ValidationError(
+                _('%(attribute_id)s must be null'), code='invalid value',
+                params={'attribute_id': self.attribute_id})
+        elif ip_addr_type == 'host':
+            is_ip_address(self.value)
+            is_unique(self.value)
+        elif ip_addr_type == 'loadbalancer':
+            is_ip_address(self.value)
+        elif ip_addr_type == 'network':
+            is_network(self.value)
+            network_overlaps(self.value, self.server.servertype_id)
```

Comment:
We now have here more or less duplicated code. Just be aware of that. ;)

Reply:
It is not fully duplicated but just partially. I am hoping to be able to remove the `intern_ip` attribute special code soon so that we have inet attributes only and then remove this code.

Reply:
Then actually a little `TODO`/comment might be nice here.